AntonioLok/online-shopping-reactJS
# add sign-up

# Changelog
- Added sign up component
- Users now should be able to create an account on my website
- Added validations on email, password fields and also async validation to check if email already exists.

# Test
- Install dependencies using `npm i`
- Start web app using `npm run start-webapp`
- Start web server using `npm run start-server`
- Click on sign up
- Register for an account 
- Note: you can test the validations by entering a non-email account on the email field, as well as entering an already created account
- Navigate to `http://localhost:8000/api/emails/:email` to check if your email has been created. emailFound: true means your email as been created, otherwise it did not get created
- If successful, it will navigate to the login page, otherwise it will throw an error.

# To do after
- Work on the login or finishing populating the products page

# Note
- I did not create a reusable form component as I felt like it would not help that much since I would be passing the fields as props to the reusable component which felt the same as if I was creating the form from scratch. I will explain more about this during the 1-1.
- I figured out why the validation was not working prev. It was because I installed a lower version of redux form so all I had to do was do an upgrade.... Now I see the importance of removing the caret in the package.json file.
- I had a lot of fun doing the async validation to check if email was already created. I always wanted to do this before but never knew how, glad it was easy with redux form.
- I put in practice what I learned from the lifion project and created my own HOC for buttons, which returns a button with icon if an icon is passed otherwise it will return a normal button. This will come in handy as I all I need to do now on is pass icon as props to the HOC if I want a icon button.
- Random: just realized that redux form wraps my entire form in a Higher Order Component... never realized I've been using HOC this whole time.
